pkg/controller/import-controller.go

```diff
@@ -72,6 +93,56 @@ func (ic *ImportController) findImportPodFromCache(pvc *v1.PersistentVolumeClaim
 	return podList[0], nil
 }
 
+func (ic *ImportController) importerRequiresScratchSpace(pvc *v1.PersistentVolumeClaim) bool {
+	contentType := getContentType(pvc)
+	// All archive requires scratch space.
+	if contentType == "archive" {
+		glog.V(3).Infof("Archive content type, requires scratch space\n")
+		return true
+	}
+	source := getSource(pvc)
+	// Invalid Sources will have gotten rejected by the web hook, so this is fine.
+	if source != SourceHTTP && source != SourceNone {
```

Comment:
As does http with specific archiving as well. Thing is, we don't know ahead of time, so we launch the importer, it sees it requires scratch space, and exits with the special exit code, which the controller sees, and it will relaunch the pod with scratch space. This check is simply to short circuit that loop for ones we know ahead of time will require scratch space.

Reply:
got you.